parts/linux/cloud-init/artifacts/ensure_imds_restriction.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+# Ensure IMDS restriction rule is correctly set on the node, this script will be registered as kubelet service's ExecStart
+# to ensure desired iptables rule will be ensured after the node is restarted.
+# Below 3 variables are passed from kubelet service's drop-in file.
+primaryNicIP="${PRIMARY_NIC_IP}"
+insertRuleToMangleTable="${INSERT_IMDS_RESTRICTION_RULE_TO_MANGLE_TABLE}"
+enableIMDSRestriction="${ENABLE_IMDS_RESTRICTION}"
+
+echo "Primary NIC IP: $primaryNicIP"
+echo "Insert IMDS restriction rule to mangle table: $insertRuleToMangleTable"
+echo "Enable IMDS restriction: $enableIMDSRestriction"
+
+# If any required variable is not set, exit with error
+if [[ -z "$primaryNicIP" || -z "$insertRuleToMangleTable" || -z "$enableIMDSRestriction" ]]; then
+    echo "One or more required variables are not set, exiting..."
+    exit 1
+fi
+
+ensureIMDSRestrictionRule() {
+    if [[ "{$insertRuleToMangleTable,,}" == "true" ]]; then
```

Comment:
sorry I must've typo'd in my original suggestion, it should be `"${insertRuleToMangleTable,,}"`

Reply:
ah, good catch, I even hasn't noticed it, fixed typo.